kentik_api_library/tests/integration/test_users.py

```diff
@@ -0,0 +1,302 @@
+    # given
+    httpretty.register_uri(
+        httpretty.GET,
+        f"{FAKE_API_URL}/user/{DUMMY_USER_ID}",
+        status=HTTPStatus.INTERNAL_SERVER_ERROR.value,
+        body='{"error":"Internal Server Error"}',
+    )
+
+    # when
+    with pytest.raises(KentikAPIError) as exc_info:
+        kentik_api.users.get(DUMMY_USER_ID)
+
+    # then
+    assert len(httpretty.latest_requests()) == 1
+    assert len(httpretty.last_request().querystring) == 0
+    assert httpretty.last_request().headers.get(AUTH_EMAIL_KEY) == DUMMY_AUTH_EMAIL
+    assert httpretty.last_request().headers.get(AUTH_API_TOKEN_KEY) == DUMMY_TOKEN
+    assert len(httpretty.last_request().body) == 0
+
+    assert str(exc_info.value) == '{"error":"Internal Server Error"}'
+
+
+@httpretty.activate
+def test_get_user_fails_when_status_service_unavailable_received(kentik_api) -> None:
+    # given
+    httpretty.register_uri(
+        httpretty.GET,
+        f"{FAKE_API_URL}/user/{DUMMY_USER_ID}",
+        status=HTTPStatus.SERVICE_UNAVAILABLE.value,
+        body='{"error":"Service Unavailable"}',
+    )
+
+    # when
+    with pytest.raises(UnavailabilityError) as exc_info:
+        kentik_api.users.get(DUMMY_USER_ID)
+
+    # then
+    assert len(httpretty.latest_requests()) == 1
+    assert len(httpretty.last_request().querystring) == 0
+    assert httpretty.last_request().headers.get(AUTH_EMAIL_KEY) == DUMMY_AUTH_EMAIL
+    assert httpretty.last_request().headers.get(AUTH_API_TOKEN_KEY) == DUMMY_TOKEN
+    assert len(httpretty.last_request().body) == 0
+
+    assert str(exc_info.value) == '{"error":"Service Unavailable"}'
+
+
+@httpretty.activate
+def test_get_user_fails_when_gateway_timeout_received(kentik_api) -> None:
+    # given
+    httpretty.register_uri(
+        httpretty.GET,
+        f"{FAKE_API_URL}/user/{DUMMY_USER_ID}",
+        status=HTTPStatus.GATEWAY_TIMEOUT.value,
+        body='{"error":"Gateway Timeout"}',
+    )
+
+    # when
+    with pytest.raises(UnavailabilityError) as exc_info:
+        kentik_api.users.get(DUMMY_USER_ID)
+
+    # then
+    assert len(httpretty.latest_requests()) == 1
+    assert len(httpretty.last_request().querystring) == 0
+    assert httpretty.last_request().headers.get(AUTH_EMAIL_KEY) == DUMMY_AUTH_EMAIL
+    assert httpretty.last_request().headers.get(AUTH_API_TOKEN_KEY) == DUMMY_TOKEN
+    assert len(httpretty.last_request().body) == 0
+
+    assert str(exc_info.value) == '{"error":"Gateway Timeout"}'
+
+
+@httpretty.activate
+def test_get_user_fails_when_invalid_response_body_received(kentik_api) -> None:
+    # given
+    httpretty.register_uri(
+        httpretty.GET,
+        f"{FAKE_API_URL}/user/{DUMMY_USER_ID}",
+        status=HTTPStatus.OK.value,
+        body='"{"dummy":"invalid response body"}"',
+    )
+
+    # when
+    # TODO(dfurman): transform implementation to use conversions.from_json wrapper
+    with pytest.raises(json.JSONDecodeError):
```

Comment:
hmm, this should actually raise DeserializationError
?

Reply:
I am currently working on transforming implementation to use conversions.from_json wrapper and it will be raising DeserializationError

Reply:
I agree, so I added a TODO. @lukaa12 is working on resolving that right now.